core/app/models/spree/notification_dispatch/action_mailer_dispatch.rb

```diff
@@ -0,0 +1,55 @@
+# Default sender class for Spree::NotificationDispatch, sends via
+# ActionMailer.
+#
+# You can see the mapping from notification message types to
+# ActionMailer classes and methods in `.mailer_dispatch_table`, which
+# you can also customize to map to different mailers or methods.
+#
+# The particular argument list for each message type is
+# specified in NotificationDispatch.signatures
+class Spree::NotificationDispatch::ActionMailerDispatch
+  class_attribute :mailer_dispatch_table
+  # values are pairs of fully qualified mailer class names, and method names
+  self.mailer_dispatch_table = {
+    carton_shipped: ['Spree::CartonMailer', :shipped_email],
+    order_confirm: ['Spree::OrderMailer', :confirm_email],
+    order_cancel: ['Spree::OrderMailer', :cancel_email],
+    order_inventory_cancel: ['Spree::OrderMailer', :inventory_cancellation_email],
+    reimbursement_processed: ['Spree::ReimbursementMailer', :reimbursement_email]
+  }
```

Comment:
I believe this is the core of the entire matter. For every event, we want to have a class and a method that's fired with arguments. Can we reduce this PR to a simple dispatcher that calls `configured_method` on `ConfiguredClass`, and remove the ActionMailerDispatch class and much of the `lambda` stuff that IMO knows too much about the actual mailers?